libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -50,8 +48,20 @@ public Task ContinueConversation(string botAppId, ConversationReference referenc
             if (callback == null)
                 throw new ArgumentNullException(nameof(callback)); 
 
-            var context = new BotFrameworkTurnContext(botAppId, this, reference.GetPostToBotMessage());
-            return RunPipeline(context, callback);
+            var context = new TurnContext(this, reference.GetPostToBotMessage());
+
+            // Hand craft Claims Identity.
+            ClaimsIdentity claimsIdentity = new ClaimsIdentity(new List<Claim>
+            {
+                // Adding claims for both Emulator and Channel.
+                new Claim(AuthenticationConstants.AudienceClaim, botAppId),
+                new Claim(AuthenticationConstants.AppIdClaim, botAppId)
+            });
+
+            context.Services.Add<IIdentity>(claimsIdentity);
+            IConnectorClient connectorClient = await this.CreateConnectorClientAsync(claimsIdentity, reference.ServiceUrl);
```

Comment:
We should be consistent in our use of `var` vs explicit types. Right now the codebase follows a `var`-only approach so, religious debates aside, I think we should stick with that.

Reply:
wildo.